packages/core/src/sdJwt/disclosures.ts

```diff
@@ -80,19 +87,22 @@ export class Disclosure {
             return this as unknown as WithCalculateDigestReturnType<HasherImplementation>
         }
 
-        // Calculate digest
-        const hashResult = hasher(this.encoded)
+        const digestResult = disclosureCalculateDigest(
+            this.asJson(),
+            // TODO: string vs HasherAlgorithm
```

Comment:
Could you elaborate in this comment or link to an issue? Not too sure what we should fix here.